x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/utils.ts

```diff
@@ -256,8 +254,6 @@ export const getTupleDuplicateErrorsAndUniqueRules = (
             );
           }
           acc.rulesAcc.set(ruleId, parsedRule);
-        } else {
```

Comment:
See why it would seem this is never hit when running end to end tests since it stops you from creating a request without `rule_id`, but can hit it on unit tests. May be good to keep as it is * possibly * an attainable case.

Reply:
Good call. I added it back here https://github.com/elastic/kibana/pull/58274/commits/640d0411bcd36db0ba889c37bdde9f8938faacd9 and added an additional unit test to show the validation on the route when ruleId is missing from rules on import.